addons/io_hubs_addon/components/definitions/physics_shape.py

```diff
@@ -0,0 +1,118 @@
+        default="all")
+
+    halfExtents: FloatVectorProperty(
+        name="Half Extents",
+        description="Half dimensions of the collider. (Only used when fit is set to \"manual\" and type is set to \"box\")",
+        unit='LENGTH',
+        subtype="XYZ",
+        default=(0.5, 0.5, 0.5))
+
+    minHalfExtent: FloatProperty(
+        name="Min Half Extent",
+        description="The minimum size to use when automatically generating half extents. (Only used when fit is set to \"all\" and type is set to \"box\")",
+        unit="LENGTH",
+        default=0.0)
+
+    maxHalfExtent: FloatProperty(
+        name="Max Half Extent",
+        description="The maximum size to use when automatically generating half extents. (Only used when fit is set to \"all\" and type is set to \"box\")",
+        unit="LENGTH",
+        default=1000.0)
+
+    sphereRadius: FloatProperty(
+        name="Sphere Radius",
+        description="Radius of the sphere collider. (Only used when fit is set to \"manual\" and type is set to \"sphere\")",
+        unit="LENGTH", default=0.5)
+
+    offset: FloatVectorProperty(
+        name="Offset", description="An offset to apply to the collider relative to the object's origin",
+        unit='LENGTH',
+        subtype="XYZ",
+        default=(0.0, 0.0, 0.0))
+
+    includeInvisible: BoolProperty(
+        name="Include Invisible",
+        description="Include invisible objects when generating a collider. (Only used if \"fit\" is set to \"all\")",
+        default=False)
+
+    def draw(self, context, layout, panel):
+        layout.prop(self, "type")
+        layout.prop(self, "fit")
+        if self.fit == "manual":
+            if self.type == "box":
+                layout.prop(self, "halfExtents")
+            elif self.type == "sphere":
+                layout.prop(self, "sphereRadius")
+        else:
+            if self.type == "box":
+                layout.prop(self, "minHalfExtent")
+                layout.prop(self, "maxHalfExtent")
+            layout.prop(self, "includeInvisible")
+        layout.prop(self, "offset")
+
+        if self.fit == "manual" and (self.type == "mesh" or self.type == "hull"):
+            col = layout.column()
+            col.alert = True
+            col.label(
+                text="'Hull' and 'Mesh' do not support 'manual' fit mode", icon='ERROR')
+
+    def gather(self, export_settings, object):
+        props = super().gather(export_settings, object)
+        props['offset'] = {
+            'x': self.offset[0],
+            'y': self.offset[2] if export_settings['gltf_yup'] else self.offset[1],
+            'z': self.offset[1] if export_settings['gltf_yup'] else self.offset[2],
+        }
+        props['halfExtents'] = {
+            'x': self.halfExtents[0],
+            'y': self.halfExtents[2] if export_settings['gltf_yup'] else self.halfExtents[1],
+            'z': self.halfExtents[1] if export_settings['gltf_yup'] else self.halfExtents[2],
```

Comment:
Nit: I believe self.offset and self.halfExtents XYZ values could be specified by .x .y .z instead of by index, which would make things a little clearer.